components/core/src/compressor_frontend/OutputBuffer.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef COMPRESSOR_FRONTEND_OUTPUT_BUFFER_HPP
+#define COMPRESSOR_FRONTEND_OUTPUT_BUFFER_HPP
+
+// Project Headers
+#include "Buffer.hpp"
+#include "Token.hpp"
+
+/**
+ * A buffer containing the tokenized output of the parser.
```

Comment:
* This comment should be word-wrapped at 80 characters.
* These details are somewhat useful but don't really seem to be handled within the class, so I would either move some of the logic into the class or move some of these comments to where they're actually relevant. My preference would be the first one to avoid cases where you might a timestamp but forget to set `m_has_timestamp`.